runtime/worker_bootstrap.rs

```diff
@@ -58,28 +57,142 @@ impl Default for BootstrapOptions {
 }
 
 impl BootstrapOptions {
-  pub fn as_json(&self) -> serde_json::Value {
-    json!({
-      // Shared bootstrap args
-      "args": self.args,
-      "cpuCount": self.cpu_count,
-      "debugFlag": self.debug_flag,
-      "denoVersion": self.runtime_version,
-      "locale": self.locale,
-      "location": self.location,
-      "noColor": self.no_color,
-      "isTty": self.is_tty,
-      "tsVersion": self.ts_version,
-      "unstableFlag": self.unstable,
-      // Web worker only
-      "enableTestingFeaturesFlag": self.enable_testing_features,
-      // Env values
-      "pid": std::process::id(),
-      "ppid": ppid(),
-      "target": env!("TARGET"),
-      "v8Version": deno_core::v8_version(),
-      "userAgent": self.user_agent,
-      "inspectFlag": self.inspect,
-    })
+  pub fn as_v8<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+  ) -> v8::Local<'s, v8::Array> {
+    let array = v8::Array::new(scope, 17);
+
+    {
+      let args = v8::Array::new(scope, self.args.len() as i32);
+      for (idx, arg) in self.args.iter().enumerate() {
+        let arg_str = v8::String::new(scope, arg).unwrap();
+        args.set_index(scope, idx as u32, arg_str.into());
+      }
+      array.set_index(scope, 0, args.into());
+    }
+
+    {
+      let val = v8::Integer::new(scope, self.cpu_count as i32);
+      array.set_index(scope, 1, val.into());
+    }
+
+    {
+      let val = v8::Boolean::new(scope, self.debug_flag);
+      array.set_index(scope, 2, val.into());
+    }
+
+    {
+      let val = v8::String::new_from_one_byte(
+        scope,
+        self.runtime_version.as_bytes(),
+        v8::NewStringType::Normal,
+      )
+      .unwrap();
+      array.set_index(scope, 3, val.into());
+    }
+
+    {
+      let val = v8::String::new_from_one_byte(
+        scope,
+        self.locale.as_bytes(),
+        v8::NewStringType::Normal,
+      )
+      .unwrap();
+      array.set_index(scope, 4, val.into());
+    }
+
+    {
+      let val: v8::Local<v8::Value> = if let Some(location) = &self.location {
+        v8::String::new_from_one_byte(
+          scope,
+          location.as_str().as_bytes(),
+          v8::NewStringType::Normal,
+        )
+        .unwrap()
+        .into()
```

Comment:
I should probably use `v8::String::new()` here instead - there's no guarantee that the location URL will not contain UTF-8